src/sinks/util/service.rs

```diff
@@ -84,137 +85,188 @@ impl<L> ServiceBuilderExt<L> for ServiceBuilder<L> {
     fn default() -> Self {
-        Self::new(CONCURRENCY_DEFAULT)
+        Self {
+            concurrency: default_concurrency(),
+            timeout_secs: default_timeout_secs(),
+            rate_limit_duration_secs: default_rate_limit_duration_secs(),
+            rate_limit_num: default_rate_limit_num(),
+            retry_attempts: default_retry_attempts(),
+            retry_max_duration_secs: default_retry_max_duration_secs(),
+            retry_initial_backoff_secs: default_retry_initial_backoff_secs(),
+            adaptive_concurrency: AdaptiveConcurrencySettings::default(),
+        }
     }
 }
 
 impl TowerRequestConfig {
-    pub const fn new(concurrency: Concurrency) -> Self {
+    pub fn new(concurrency: Concurrency) -> Self {
         Self {
             concurrency,
-            timeout_secs: Some(TIMEOUT_SECONDS_DEFAULT),
-            rate_limit_duration_secs: Some(RATE_LIMIT_DURATION_SECONDS_DEFAULT),
-            rate_limit_num: Some(RATE_LIMIT_NUM_DEFAULT),
-            retry_attempts: Some(RETRY_ATTEMPTS_DEFAULT),
-            retry_max_duration_secs: Some(RETRY_MAX_DURATION_SECONDS_DEFAULT),
-            retry_initial_backoff_secs: Some(RETRY_INITIAL_BACKOFF_SECONDS_DEFAULT),
-            adaptive_concurrency: AdaptiveConcurrencySettings::const_default(),
+            ..Default::default()
         }
     }
 
-    pub const fn const_default() -> Self {
-        Self::new(CONCURRENCY_DEFAULT)
-    }
-
     pub const fn timeout_secs(mut self, timeout_secs: u64) -> Self {
-        self.timeout_secs = Some(timeout_secs);
+        self.timeout_secs = Duration::from_secs(timeout_secs);
         self
     }
 
     pub const fn rate_limit_duration_secs(mut self, rate_limit_duration_secs: u64) -> Self {
-        self.rate_limit_duration_secs = Some(rate_limit_duration_secs);
+        self.rate_limit_duration_secs = Duration::from_secs(rate_limit_duration_secs);
         self
     }
 
     pub const fn rate_limit_num(mut self, rate_limit_num: u64) -> Self {
-        self.rate_limit_num = Some(rate_limit_num);
+        self.rate_limit_num = rate_limit_num;
         self
     }
 
     pub const fn retry_attempts(mut self, retry_attempts: usize) -> Self {
-        self.retry_attempts = Some(retry_attempts);
+        self.retry_attempts = retry_attempts;
         self
     }
 
     pub const fn retry_max_duration_secs(mut self, retry_max_duration_secs: u64) -> Self {
-        self.retry_max_duration_secs = Some(retry_max_duration_secs);
+        self.retry_max_duration_secs = Duration::from_secs(retry_max_duration_secs);
         self
     }
 
     pub const fn retry_initial_backoff_secs(mut self, retry_initial_backoff_secs: u64) -> Self {
-        self.retry_initial_backoff_secs = Some(retry_initial_backoff_secs);
+        self.retry_initial_backoff_secs = Duration::from_secs(retry_initial_backoff_secs);
         self
     }
 
     pub fn unwrap_with(&self, defaults: &Self) -> TowerRequestSettings {
```

Comment:
The changes in this function feel like they could be done more cleanly, but nothing is coming to mind right now.

Reply:
The whole usage of this function is pretty strange code flow.
I started down the path of further refactoring it, but reigned myself in as it began to cascade into more and more changes.

Reply:
I think you might not need this function now? If we deserialize, then the defaults will already be baked in. If we form an instance of `TowerRequestSettings` by hand, we can provide the defaults with `..Default::default()`.

Reply:
collapsing this one as discussion converged here: https://github.com/vectordotdev/vector/pull/16150#discussion_r1088417421